### PR TITLE
Refact test-td-payload in tests

### PR DESCRIPTION
**Pull request by liuw1**
Implement a simple test framework to run test cases with TDX dependency.

This PR depends on [PR 130](https://github.com/confidential-containers/td-shim/pull/130). It will use latest enroll tool to enroll the test json file in CFV. 

Signed-off-by: Wei Liu <wei3.liu@intel.com>

### CODE REVIEW

File: doc/test_with_td_payload.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Test with TD Payload
+TD-Shim has some test cases rely on TDX specific environment. Therefore, this simple test framework is created to support the test.
+
+## Test framework code structure
+All codes of test framework are located in [tests/td-test-payload](../tests/test-td-payload).
```
Comment (jiangliu) [inline doc/test_with_td_payload.md:5]: tests/test-td-payload?

File: tests/test-td-payload/Cargo.toml [hunk 2]
```diff
@@ -1,26 +1,30 @@
 [package]
 name = "test-td-payload"
 version = "0.1.0"
-description = "Run TDX hardware specific unit test cases inside a VM"
```
Comment (jiangliu) [inline tests/test-td-payload/Cargo.toml:4]: It would be better to keep these lines.

Comment (liuw1) [reply tests/test-td-payload/Cargo.toml:4]: Agree

File: doc/test_with_td_payload.md (hunk 1 above)
Comment (liuw1) [reply doc/test_with_td_payload.md:5]: Thank you, I will correct it.

**Comment by jiangliu** [comment]
This test framework should also work on normal x86 hardware when the tdx feature is disabled, so could we test it in the  github integration workflow with tdx disabled?

**Comment by liuw1** [comment]
> This test framework should also work on normal x86 hardware when the tdx feature is disabled, so could we test it in the github integration workflow with tdx disabled?

This test framework created only for testing TDX related features, it can only works when the TDX feature is enabled.

**Pull request merged**